wallet/src/Pos/Wallet/Web/Methods/Logic.hs

```diff
@@ -219,6 +221,14 @@ getWallets = do
     mps <- withTxpLocalData getMempoolSnapshot
     mapM (getWalletIncludeUnready ws mps False) (getWalletAddresses ws)
 
+getWalletsWithInfo ::  MonadWalletLogicRead ctx m => m [(CWallet, WalletInfo)]
+getWalletsWithInfo = do
+    ws <- askWalletSnapshot
```

Comment:
Same deal: if this is not a top level handler we might want to pass both `mps` and `ws` externally.